chia/wallet/puzzles/dao_update_proposal.clsp

```diff
@@ -1,3 +1,5 @@
+; [REVIEW - misc] I think these files should be grouped into a folder or have a prefix or something to classify them as potential proposals
+; [REVIEW - optimization] That said, I don't think this needs to be a file at all.  Just `(1, ((CREATE_COIN puzzle_hash_that_can_be_calculated_and_verified_off_chain ONE)))`.
```

Comment:
as you've identified elsewhere in your review, we sacrifice optimisation for the sake of letting non-technical users have a reasonably safe integrated experience. Having known proposal types that can be parsed by the wallet that are revealed on chain is the main one of these

Reply:
So as a user, what am I going to be getting/checking here?  My understanding is this:
1) I detect a proposal to vote on, which has a curried in dpuz hash
2) I get some information separately that the mod hash is this file and what the curried args are
3) I validate the args are good to me
4) I curry the args into the mod hash and check that it matches the proposed dpuz

But for some reason after all of that validation, we make full nodes do the tree hash of the new DAO for us.  What I'm proposing is:
1) I detect a proposal to vote on, which has a curried in dpuz hash
2) I get some information separately that it's an update proposal with the known form  `(1, ((CREATE_COIN new_dao_hash ONE)))` and what the curried args are
3) I validate the args are good to me
4) I curry the args into the treasury mod hash, stick it in the format above, and make sure that matches the proposed dpuz

It's not anymore complex, but you don't have to do the `puzzle-hash-of-a-curried-function`.